app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,26 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    name = params[:user_name]
+    user = User.find_by(name: name)
+
+    if user
+      session[:user_id] = user.id
+      flash[:success] = "#{user.name} is successfully logged in"
+      redirect_to root_path
+
+    else
+      user = User.create(name: name)
+      redirect_to root_path
+      flash[:success] = "#{user.name} is successfully logged in"
```

Comment:
When you create a new user you're not saving their ID in the session. That means that a new user has to enter their name twice to log in: once to create the record, and once to set their user ID in the session.